src/metabase/core.clj

```diff
@@ -133,6 +145,10 @@
     ;; start the metabot thread
     (metabot/start-metabot!))
 
+  (sync-locale)
+  (future (while true (do (Thread/sleep 1000) (sync-locale))))
```

Comment:
What's the goal here?

Reply:
This is a bit of a hack to keep the system locale in sync with the setting. See https://github.com/metabase/metabase/pull/5803/files#diff-a61f704fe001e6dad2115909432f583dR40 I tried adding an event listener like we have for metabot (for example) but couldn't get it working for some reason.

Reply:
@camsaul I took another look at this and it looks like we only publish the `settings-update` event for `set-many!` but not `set!` etc, as mentioned here https://github.com/metabase/metabase/blob/master/src/metabase/models/setting.clj#L381

I feel like it would actually be useful for that event to fire for all the other `set*!` functions too. It doesn't seem like there would be another easy way to be notified when the locale setting changes (aside from polling like I've done so far). If I followed the suggestion in the TODO linked above, I'd have to add a custom endpoint and a special case in the frontend to use that endpoint for the `site-locale` setting (or a special case in the `PUT /api/setting/:key` endpoint)

Reply:
you don't need to do any of that stuff. You can give Settings customer magic setter functions. Take a look at `site-url` in `public_settings.clj` for an example. 

So here just give your `site-locale` setting a custom setter that calls that `sync-locale` function.

Reply:
btw doing something like this is a bad idea because

1.  it's going to start calling `sync-locale` as soon as the namespace is evaluated, which means (if I understand what I think it's supposed to be doing) it will possibly be trying to use Settings before the DB is initialized (race condition). If the DB isn't ready yet it will throw an Exception that you won't end up seeing because it's on a background thread.
2. When that Exception gets thrown (or any other exception is raised by `sync-locale`)  the loop will break and `sync-locale` won't be called any additional times.

At any rate you don't need to schedule a recurring task here, but if you did, we have Quartz and the `metabase.task` stuff for that instead of reïnventing the wheel

Reply:
Ok, I'll use a custom setting setter.

The `sync-locale` thing was just a temporary thing to get it working.

Reply:
Done.